taf/tests/test_updater/update_utils.py

```diff
@@ -256,3 +257,112 @@ def _update_expect_error():
                 assert client_repository.path.exists()
             else:
                 assert not client_repository.path.exists()
+
+
+def check_repo_sync(client_dir, origin_auth_repo):
+    """
+    Asserts that the repositories in the client directory are updated
+    to match the commits in the origin authentication repository.
+
+    Args:
+        client_dir (Path): Path to the client directory.
+        origin_auth_repo (AuthenticationRepository): The origin authentication repository.
+    """
+    # Load the target repositories from the origin auth repo
+    client_repos = {
+        repo.name: GitRepository(client_dir, repo.name)
+        for repo in load_target_repositories(origin_auth_repo).values()
+    }
+    # Add the origin authentication repository itself to the client repositories
+    client_repos[origin_auth_repo.name] = GitRepository(
+        client_dir, origin_auth_repo.name
+    )
+
+    try:
```

Comment:
Catching assertion errors is not considered to be good practice. If you need to reuse the check, create a helped function without an assert.